src/features/query-bar/components/filter/Condition.js

```diff
@@ -58,88 +56,84 @@ const Condition = ({
     columns,
     condition,
     deleteCondition,
+    onFieldChange,
     index,
     selectedConnector,
-    update,
     onConnectorChange,
 }: Props) => {
     const onDeleteButtonClick = () => {
         deleteCondition(index);
     };
 
-    const updateSelectedField = (option: SelectOptionType, fieldType?: string) => {
-        const conditionIndex = index;
-        const value = option.value;
-        const valueType = option.type;
-        const fieldId = option.fieldId || condition.fieldId;
+    const updateSelectedField = (option: OptionType, fieldType?: string) => {
+        const { displayText, type, value } = option;
 
-        const displayText = option.displayText;
         let displayTextType = '';
         let keyType = '';
 
-        if (fieldType === ATTRIBUTE) {
-            displayTextType = COLUMN_DISPLAY_TEXT;
-            keyType = COLUMN_KEY;
-        } else if (fieldType === OPERATOR) {
-            displayTextType = OPERATOR_DISPLAY_TEXT;
-            keyType = OPERATOR_KEY;
-        } else if (fieldType === VALUE) {
-            displayTextType = VALUE_DISPLAY_TEXT;
-            keyType = VALUE_KEY;
+        switch (fieldType) {
+            case COLUMN:
+                displayTextType = COLUMN_DISPLAY_TEXT;
+                keyType = COLUMN_KEY;
+                break;
+            case OPERATOR:
+                displayTextType = OPERATOR_DISPLAY_TEXT;
+                keyType = OPERATOR_KEY;
+                break;
+            case VALUE:
+                displayTextType = VALUE_DISPLAY_TEXT;
+                keyType = VALUE_KEY;
+                break;
+            default:
+                throw new Error('invalid input');
         }
 
-        update(conditionIndex, condition, displayText, displayTextType, fieldId, value, keyType, valueType);
+        onFieldChange(index, condition, displayText, displayTextType, value, keyType, type);
     };
 
     const getFormattedOptions = (options: Array<Object>): any[] => {
         return options.map(option => {
+            const { displayName, type } = option;
             return {
-                displayText: option.displayName,
-                fieldId: option.id,
-                type: option.type,
-                value: option.displayName,
+                displayText: displayName,
```

Comment:
Too bad the names are a bit inconsistent.

Reply:
I checked with Loren regarding changing displayName to be displayText in the Column object but it looks like it'll involve changing 12~ files on both of our ends.